tests/requestparsing/header_test.go

```diff
@@ -0,0 +1,197 @@
+package requestparsing
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/google/go-safeweb/testing/requesttesting"
+)
+
+var headerTests = []struct {
+	name    string
+	request []byte
+	want    map[string][]string
+}{
+	{
+		name: "Basic",
+		// All tests use verbatim newline characters instead of using multiline strings to ensure that \r and \n end up in exactly the right places.
```

Comment:
This is uncomfortably long, please add some linebreaks. Also, consider adding the comment a few levels higher instead of by the first test case.